src/util/pg.py

```diff
@@ -196,27 +196,58 @@ def explode_query_range(cr, query, table, alias=None, bucket_size=10000, prefix=
 
     alias = alias or table
 
-    cr.execute("SELECT min(id), max(id) FROM {}".format(table))
+    if "{parallel_filter}" not in query:
+        sep_kw = " AND " if re.search(r"\sWHERE\s", query, re.M | re.I) else " WHERE "
+        query += sep_kw + "{parallel_filter}"
+
+    cr.execute(format_query(cr, "SELECT min(id), max(id) FROM {}", table))
     min_id, max_id = cr.fetchone()
     if min_id is None:
         return []  # empty table
+    count = (max_id + 1 - min_id) // bucket_size
+    if count > 100000:
```

Comment:
As it is to avoid potential MemoryError, it depends on the machine it runs on.
I would do it like `MAX_WORKERS` and define this threshold behind an env variable.
```suggestion
    if count > int(os.getenv("MAX_BUCKETS", "100000")):
```

Reply:
I increased the value to 150K -> found a case in our logs with 129K that ran without issues (relatively fast). cc: @cawo-odoo